master/internal/run/runs.go

```diff
@@ -61,8 +61,9 @@ func flattenMetadata(data map[string]any) (flatMetadata []model.RunMetadataIndex
 		key    string
 		value  any
 		depth  int
+		array  bool
 	}
-	stack := []metadataEntry{{prefix: "", key: "", value: data, depth: 0}}
+	stack := []metadataEntry{{prefix: "", key: "", value: data, depth: 0, array: false}}
```

Comment:
question: The intended behavior here is to trail down the array property to any nested structures? Does that mean we're only allowing filtering/sorting on metadata that is not inside of an array, absolutely okay if that's the case, just wanted to make sure I was understanding

Reply:
Yes. The only exception would be the keys of an object that is inside of an array eg.
```
{
    "key1": ["value"] <--- "key1" is an array type
    "key2": [{"nested": "value"}] <--- "key2.nested" is string type as it can be filtered/sorted
}
```